src/auth/src/credentials.rs

```diff
@@ -48,24 +48,77 @@ type Result<T> = std::result::Result<T, crate::errors::CredentialError>;
 /// [Metadata Service]: https://cloud.google.com/compute/docs/metadata/overview
 /// [Google Compute Engine]: https://cloud.google.com/products/compute
 /// [Google Kubernetes Engine]: https://cloud.google.com/kubernetes-engine
-pub trait Credential: Send + Sync {
-    /// Asynchronously retrieves a token.
-    ///
-    /// Returns a [Token][crate::token::Token] for the current credentials.
-    /// The underlying implementation refreshes the token as needed.
-    fn get_token(&mut self) -> impl Future<Output = Result<crate::token::Token>> + Send;
+pub struct Credential {
+    inner: Box<dyn traits::dynamic::Credential>,
+}
+
+impl traits::Credential for Credential {
+    fn get_token(&mut self) -> impl Future<Output = Result<crate::token::Token>> + Send {
+        self.inner.get_token()
+    }
 
-    /// Asynchronously constructs the auth headers.
-    ///
-    /// Different auth tokens are sent via different headers. The
-    /// [Credential] constructs the headers (and header values) that should be
-    /// sent with a request.
-    ///
-    /// The underlying implementation refreshes the token as needed.
     fn get_headers(
         &mut self,
-    ) -> impl Future<Output = Result<Vec<(HeaderName, HeaderValue)>>> + Send;
+    ) -> impl Future<Output = Result<Vec<(HeaderName, HeaderValue)>>> + Send {
+        self.inner.get_headers()
+    }
+
+    fn get_universe_domain(&mut self) -> impl Future<Output = Option<String>> + Send {
+        self.inner.get_universe_domain()
+    }
+}
+
+pub mod traits {
+    use super::Future;
+    use super::Result;
+    use super::{HeaderName, HeaderValue};
+
+    pub trait Credential {
```

Comment:
This needs a comment, it is a public API.  Try `cargo doc` to see the documents locally.